linux/Platform.c

```diff
@@ -750,111 +728,97 @@ static void Platform_Battery_getProcData(double* percent, ACPresence* isOnAC) {
 // ----------------------------------------
 
 static void Platform_Battery_getSysData(double* percent, ACPresence* isOnAC) {
+    unsigned long int   totalFull = 0, totalRemain = 0;
+    bool                full = false, now = false;
+    int                 fullCharge = 0,val = 0;
+    char                filePath[256], *buf = NULL, buffer[1024], field[100];
+    double              capacityLevel = NAN;
+    ssize_t             r = 0;
+    
+    *percent = NAN;
+    *isOnAC = AC_ERROR;
+
+    DIR* dir = opendir(SYS_POWERSUPPLY_DIR);
+    if (!dir)
+        return;
+
+
+    for (;;) {
+        const struct dirent* dirEntry = readdir(dir);
+
+        const char* line;
+        if (!dirEntry)
+            break;
 
-   *percent = NAN;
-   *isOnAC = AC_ERROR;
-
-   DIR* dir = opendir(SYS_POWERSUPPLY_DIR);
-   if (!dir)
-      return;
-
-   unsigned long int totalFull = 0;
-   unsigned long int totalRemain = 0;
-
-   for (;;) {
-      const struct dirent* dirEntry = readdir(dir);
-      if (!dirEntry)
-         break;
-
-      const char* entryName = dirEntry->d_name;
-      char filePath[256];
-
-      xSnprintf(filePath, sizeof filePath, SYS_POWERSUPPLY_DIR "/%s/type", entryName);
-
-      char type[8];
-      ssize_t r = xReadfile(filePath, type, sizeof(type));
-      if (r < 3)
-         continue;
-
-      if (type[0] == 'B' && type[1] == 'a' && type[2] == 't') {
-         xSnprintf(filePath, sizeof filePath, SYS_POWERSUPPLY_DIR "/%s/uevent", entryName);
-
-         char buffer[1024];
-         r = xReadfile(filePath, buffer, sizeof(buffer));
-         if (r < 0) {
-            closedir(dir);
-            return;
-         }
-
-         char* buf = buffer;
-         const char* line;
-         bool full = false;
-         bool now = false;
-         int fullSize = 0;
-         double capacityLevel = NAN;
-
-         #define match(str,prefix) \
-            (String_startsWith(str,prefix) ? (str) + strlen(prefix) : NULL)
-
-         while ((line = strsep(&buf, "\n")) != NULL) {
-            const char* ps = match(line, "POWER_SUPPLY_");
-            if (!ps)
-               continue;
-            const char* capacity = match(ps, "CAPACITY=");
-            if (capacity)
-               capacityLevel = atoi(capacity) / 100.0;
-            const char* energy = match(ps, "ENERGY_");
-            if (!energy)
-               energy = match(ps, "CHARGE_");
-            if (!energy)
-               continue;
-            const char* value = (!full) ? match(energy, "FULL=") : NULL;
-            if (value) {
-               fullSize = atoi(value);
-               totalFull += fullSize;
-               full = true;
-               if (now)
-                  break;
-               continue;
+        const char* entryName = dirEntry->d_name;
+        
+        if(String_startsWith(entryName, "BAT")){
+            xSnprintf(filePath, sizeof filePath, SYS_POWERSUPPLY_DIR "/%s/uevent", entryName);
```

Comment:
I know `sizeof` in C is an operator, but the codebase consistently uses `sizeof(x)`.
```suggestion
            xSnprintf(filePath, sizeof(filePath), SYS_POWERSUPPLY_DIR "/%s/uevent", entryName);
```